File: 0198-house-robber/0198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        f=[0]*len(nums)
        for j in range(len(nums)):
            if j==0:
                f[0]=nums[0]
            elif j==1:
                f[1]=max(nums[0],nums[1])
            else:
                if f[j-1]==f[j-2]:
                    f[j]=f[j-1]+nums[j]
                else:
                    f[j]=max(f[j-2]+nums[j],f[j-1])
        return f[len(nums)-1]
                

File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        int res = nums[0];
        int n = nums.length;
        int[] maxAmount = new int[n];
        
        maxAmount[0] = nums[0];
        
        for (int i = 1; i < n; i++){   
            maxAmount[i] = nums[i];
            int temp = 0;
            for (int j = 0; j < i - 1; j++)
                temp = Math.max(temp, maxAmount[j]);

            res = Math.max(res, maxAmount[i] + temp);
            maxAmount[i] = res;
        }
        
        return res;
    }
}